src/components/AvatarWithIndicator.js

```diff
@@ -59,12 +59,15 @@ const AvatarWithIndicator = (props) => {
         isLarge ? styles.statusIndicatorLarge : styles.statusIndicator,
     ];
 
-    const hasPolicyMemberError = _.some(props.policiesMemberList, PolicyUtils.hasPolicyMemberError);
-    const hasCustomUnitsError = _.some(props.policies, PolicyUtils.hasCustomUnitsError);
-    const hasPaymentMethodError = PaymentMethods.hasPaymentMethodError(props.bankAccountList, props.cardList);
-    const hasWalletError = !_.isEmpty(props.userWallet.errors);
-    const hasAnyPolicyError = _.some(props.policies, PolicyUtils.hasPolicyError);
-    const shouldShowIndicator = hasPolicyMemberError || hasCustomUnitsError || hasPaymentMethodError || hasWalletError || hasAnyPolicyError;
+    // Check for errors and return as soon as there is one.
```

Comment:
```suggestion
    // All of the error-checking methods are put into an array. This is so that using _.some() will return
    // early as soon as the first error is returned. This makes the error checking very efficient since
    // we only care if a single error exists anywhere.
```